File: pop.js
var as = require('async');

var redis = require("redis"),
    client = redis.createClient();
client2 = redis.createClient();

var cnt = 0;

//client.SUBSCRIBE("q1", function(err) {
//})

as.forever(
    function (next) {
        client2.BRPOP('q1', 0, function (list, data) {

            if (data) {

                var f = data[1].split('').sort();
                cnt++;

                //console.log(cnt + ': ' + f.join().substring(0, 15));
            }

            next();
        });
    },
    function (err) {
        console.error(err);
        client.quit();
        process.exit(0);
    }
);

File: pop.py
import redis
import string

r = redis.Redis(host='localhost', port=6379, db=0)
con = 0

while True:
    try:
        data = r.brpop("q1")

        _a = list(data[1])
        _a.sort()
        con += 1

        if con % 50 == 0:
            print(".", end="", flush=True)

    except Exception:
        print("error")
